code/script.js

```diff
@@ -262,82 +274,133 @@ const selectQuestion = () => {
+    )
   } else {
+    alert(
+      `No, the person doesn't have ${value} hair. Removing people with ${value} hair.`
+      )
+    }
+  } else if (category === 'eye color') {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all persons with yellow hair"
-    } else {
-      // alert popup that says something like: "NO, the person doesnt have yellow hair! Remove all persons with yellow hair"
+    alert(`Yes, the person has ${value} eyes! Keeping all the people with ${value} eyes.`
+    )
+  } else {
+    alert(`No, the person doesn't have ${value} eyes. Removing people with ${value} eyes.`
+      )
     }
+  } else if (category === 'other') {
+    if (keep) {
+      alert(`Yes, the person is a ${attribute}. Keeping all the people that are ${attribute}s.`
+      )
+  } else {
+      alert(`No, the person is not a ${attribute}. Removing all the people that are ${attribute}s.`)
+    } 
   }
 
-  // filter to keep or remove based on the keep variable.
-  /* charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-    or 
-    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value) */
 
+  // filter to keep or remove based on the keep variable.
+  if (keep) {
+    //user guessed right
+    charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
+  } else {
+    //user guessed wrong
+    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+  }
+  generateBoard()
   // Invoke a function to redraw the board with the remaining people.
 }
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (suspect) => {
-  // store the interaction from the player in a variable.
+  const isConfirmed = confirm(`Are you sure you want to guess on ${suspect}?`)
+
+  if (isConfirmed) {
+    checkMyGuess(suspect)
+  }
+// store the interaction from the player in a variable.
   // remember the confirm() ?
   // If the player wants to guess, invoke the checkMyGuess function.
 }
 
 // If you confirm, this function is invoked
 const checkMyGuess = (suspect) => {
   // 1. Check if the suspect is the same as the secret person's name
+  if (suspect === secret.name) {
   // 2. Set a Message to show in the win or lose section accordingly
+    winOrLoseText.innerHTML = `You guessed on ${suspect} and it was correct!`
+  } else {
+    winOrLoseText.innerHTML = `No it's not ${suspect}. It was ${secret.name}!`
+  }
   // 3. Show the win or lose section
-  // 4. Hide the game board
+  winOrLose.style.display = 'flex'
+   // 4. Hide the game board
+  board.style.display = 'none'
 }
 
 // Invokes the start function when website is loaded
 start()
 
 // All the event listeners
+playAgainButton.addEventListener('click', start)
 restartButton.addEventListener('click', start)
+questions.addEventListener('change', () => selectQuestion())
```

Comment:
I realize this is from the last Q/A but since we no longer pass along an argument, we can write this like the others:
`questions.addEventListener('change', selectQuestion)`